packages/promotion/src/utils/compute-actions/items.ts

```diff
@@ -67,10 +68,14 @@ export function applyPromotionToItems(
         method.quantity,
         applicationMethod?.max_quantity!
       )
-      const promotionValue =
-        parseFloat(applicationMethod!.value!) * quantityMultiplier
-      const applicableTotal =
-        method.unit_price * quantityMultiplier - appliedPromoValue
+      const totalItemValue = method.unit_price * quantityMultiplier
+      let promotionValue = parseFloat(applicationMethod!.value!)
+      const applicableTotal = totalItemValue - appliedPromoValue
+
+      if (applicationMethod?.type === ApplicationMethodType.PERCENTAGE) {
+        promotionValue = (promotionValue / 100) * applicableTotal
+      }
+
```

Comment:
thought: would it make sense to move all this responsibility outside of the promotion module - instead the promotion module would expect `subtotal` fields on the items it gets passed?

Reply:
todo: use `method.subtotal / method.quantity` instead of `method.unit_price`.